deps/openlibm.version

```diff
@@ -1,2 +1,2 @@
-OPENLIBM_BRANCH=v0.4.1
-OPENLIBM_SHA1=3ee2a6e6a9fc68cbf2ab00f830b9059215e40b06
+OPENLIBM_BRANCH=v0.5.1
+OPENLIBM_SHA1=07f61f312bc97e56a32e6bf5cac7ea39fad7d4d1
```

Comment:
also need to update the checksum files, though maybe it's better if you split out the openlibm/openspecfun bumps into an adopted version of #15212, separate from the libjulia soname changes